### PR TITLE
snek5000-cbox tutorial

### PR DESCRIPTION
**Pull request by akhoubani**

### CODE REVIEW

**Comment by paugier** [comment]
Thank you @akhoubani. How long is it to execute this simulation on a small PC? If it is not too long (less than 1 min), it could be better not to have the executed notebook in the repository.

**Comment by paugier** [comment]
We also need to know if such simulations can be executed on the readthedocs servers.

**Comment by paugier** [comment]
@ashwinvis do you think it could be possible to run these simulations on RTD servers? With conda, we should be able to install gfortran so I guess it could work.

@akhoubani would it be possible to decrease a bit `params.nek.general.end_time`?

**Comment by paugier** [comment]
Is it done with a constant dt ? Could we increase a bit dt ?

**Comment by akhoubani** [comment]
> Is it done with a constant dt ? Could we increase a bit dt ?

No, dt is variable. I'll work on it to reduce the running time with increasing CFL.

**Comment by paugier** [comment]
Could you try to convert your jupyter notebook into a mardown file with [jupytext](https://jupytext.readthedocs.io/en/latest/install.html) (as a command line as described [here](https://jupytext.readthedocs.io/en/latest/using-cli.html)) and to put the .md file outside of the directory executed. Then, you should be able to use in the documentation this .md file (and you can remove the .ipynb file). With MyST, it should be executed automatically (with the result cached).

If it works, we could have only the input of your notebook in the .md. The output being produced automatically during doc building.

Could you please try that?

**Comment by akhoubani** [comment]
> MyST
How can I execute the produced .md file?

**Comment by paugier** [comment]
Isn't it automatically executed during the build? Of course, you need to use this file instead of your .ipynb file in tutorials.rst. If it does not work, could you please commit this file so that I can try to fix the problem?

**Comment by ashwinvis** [comment]
> @ashwinvis do you think it could be possible to run these simulations on RTD servers? With conda, we should be able to install gfortran so I guess it could work.

We can try. You will have to add `snek5000-cbox` at the stable branch in `docs/requirements.txt`. Btw, have you updated the stable branch?

**Comment by paugier** [comment]
Is there a way to hide all these Fortran warnings? Too much warnings are really useless because anyway, no one is going to read such long list. This is actually a general problem of snek5000...

**Comment by paugier** [comment]
@ashwinvis Strange this error. Do you understand?

```
ERROR: Cannot install snek5000 and snek5000[docs]==0.1.dev1+g9735f94 because these package versions have conflicting dependencies.

The conflict is caused by:
    snek5000[docs] 0.1.dev1+g9735f94 depends on snek5000 0.1.dev1+g9735f94 (from /home/runner/work/snek5000/snek5000)
    phill 0.3.0a1.dev27+g973bb2e depends on snek5000>=0.4.0b1
```

**Comment by ashwinvis** [comment]
Yes, you can! The way is to set warnings=False

https://github.com/snek5000/snek5000-phill/blob/973bb2e2f45324df6fbb9d398ee7c9377a278725/src/phill/Snakefile#L10

Now if you want to activate strict warnings only for the usr code. ..
There is also a hack I came up with in my abl code.
In makefile_usr.inc template

```jinja

{% for comment in comments.splitlines() -%}
# {{ comment }}
{% endfor -%}
CUSTOM_FFLAGS = {{ custom_fortran_flags }}

{# override usrfile directive only if custom flags are used -#}
{% if custom_fortran_flags != "$(FL2)" -%}
usrfile:
@env CASENAME=$(CASENAME) PPS="$(PPS)" $S/mkuserfile
$(FC) $(subst -fcheck=all,,$(CUSTOM_FFLAGS)) -Wno-unused-dummy-argument $(VISITNEK_INCLUDES) -c $(CASEDIR)/${CASENAME}.f
{% endif -%}

{% macro make(sources) %}
{# A cosmetic macro to indent comments nicely #}
{%- set source = sources[0].split("/")[-1] -%}
{%- set target = source.replace(".f", ".o") -%}
{%- set rule_string = "$(OBJDIR)/{}:{}\n\t$(F77) -c {} -I./ $< -o $@".format(
target.ljust(20),
" ".join(sources),
flags_var[source]
)
-%}
{{ rule_string }}
{% endmacro -%}

{% for sources in list_of_sources -%}
{{ make(sources) }}
{% endfor -%}
# vim: set ft=make
```

And in the Output class

```py
def write_makefile_usr(self, template, fp=None, **template_vars):
# Prepare dictionary for overriding custom fortran flags
flags_var = {
sources[0]: "$(CUSTOM_FFLAGS)"
for sources in itertools.chain.from_iterable(
self.makefile_usr_sources.values()
)
}

if os.getenv("SNEK_DEBUG"):
custom_fortran_flags = (
"$(subst -w,,$(FL2)) -fcheck=all -Wall -Wextra -Waliasing "
"-Wsurprising -Wcharacter-truncation -Wno-unused-parameter"
)

# NOTE: special overrides
flags_var["penalty_utils.f"] = custom_fortran_flags.replace(
"-fcheck=all", ""
)
else:
custom_fortran_flags = "$(FL2)"

template_vars["flags_var"] = flags_var
template_vars["custom_fortran_flags"] = custom_fortran_flags
super().write_makefile_usr(template, fp, **template_vars)
```
-------- Original Message --------

**Comment by ashwinvis** [comment]
setuptools_scm did not work leading to wrong version.

File: setup.cfg [hunk 1]
```diff
@@ -59,6 +59,8 @@ docs =
     # https://github.com/sphinx-doc/sphinx/issues/8198
     pygments >= 2.4.1
     linkify-it-py
+    phill @ git+https://github.com/snek5000/snek5000-phill.git
+    snek5000-cbox @ git+https://github.com/snek5000/snek5000-cbox.git@stable
```
Comment (ashwinvis) [inline setup.cfg:63]: I recall that packaging for pypi does not permit this. So we have to either:

1. Push phill and cbox release
1. Use docs/requirements.txt file

Comment (paugier) [reply setup.cfg]: I removed phill and snek5000-cbox from setup.cfg. We would have to upload them to PyPI.

But then, this brings the question of the name of the package phill. I think snek5000-phill would be better.

Comment (ashwinvis) [reply setup.cfg]: I agree. Let's change the name of the package. But perhaps retain the name of the top level module as `phill`. I cant imagine myself typing a long `import snek5000_phill` all the time and I dont want to give that impression that is the only way.

**Comment by paugier** [comment]
Even with `warnings=False`, I still get a lot of warnings! It would be nice to understand why...

**Comment by akhoubani** [comment]
When I use the master branch of nek5000, I don't get the warnings.

**Comment by ashwinvis** [comment]
> Even with `warnings=False`, I still get a lot of warnings! It would be nice to understand why...

I found out now that `warnings=False` invokes `append_debug_flags` function, which only does something when `SNEK_DEBUG=1`. I think I was going for the default behaviour of Nek5000, unless we are on debug mode. I can change this for better user experience.

**Comment by paugier** [comment]
I'm going to merge like this and to create following issues.

**Comment by paugier** [comment]
Thank you @akhoubani ! Nice and useful tutorial!

**Pull request merged**